src/tracks/ui/BrushHandle.cpp

```diff
@@ -280,99 +277,91 @@ UIHandle::Result BrushHandle::Drag
    auto &trackPanel = TrackPanel::Get( *pProject );
    auto &viewInfo = ViewInfo::Get( *pProject );
 
-   int x = mAutoScrolling ? mMostRecentX : event.m_x;
-   int y = mAutoScrolling ? mMostRecentY : event.m_y;
-   mMostRecentX = x;
-   mMostRecentY = y;
-
-   double posTime;
-   wxInt64 posFreq;
-   auto posToLongLong = [&](int x0){
-      posTime = viewInfo.PositionToTime(x0, mRect.x);
+   auto posXToHopNum = [&](int x0){
+      double posTime = viewInfo.PositionToTime(x0, mRect.x);
       sampleCount sc = wt->TimeToLongSamples(posTime);
-      return sc.as_long_long();
+      return sc.as_long_long() / mpSpectralData->GetHopSize();
+   };
+
+   auto posYToFreqBin = [&](int y0){
+      int resFreq = PositionToFrequency(wt, 0, y0, mRect.y, mRect.height);
+      return resFreq / (mpSpectralData->GetSR() / mpSpectralData->GetWindowSize());
    };
 
    // Clip the coordinates
    // TODO: Find ways to access the ClipParameters (for the mid)
-   int dest_xcoord = std::max(mRect.x + 10, std::min(x, mRect.x + mRect.width - 20));
-   int dest_ycoord = std::max(mRect.y + 10, std::min(y, mRect.y + mRect.height - 10));
-   int destFreq = PositionToFrequency(wt, 0, dest_ycoord, mRect.y, mRect.height);
+   int dest_xcoord = std::clamp(event.m_x, mRect.x + 10, mRect.x + mRect.width);
+   int dest_ycoord = std::clamp(event.m_y, mRect.y + 10, mRect.y + mRect.height);
 
-   int x1 = posToLongLong(dest_xcoord) / mpSpectralData->GetHopSize();
-   int y1 = destFreq / (mpSpectralData->GetSR() / mpSpectralData->GetWindowSize());
+   int h1 = posXToHopNum(dest_xcoord);
+   int b1 = posYToFreqBin(dest_ycoord);
 
    mbCtrlDown = event.ControlDown();
 
    // Use the hop and bin number to calculate the brush stroke, instead of the mouse coordinates
    // For mouse coordinate:
    // (src_xcoord, src_ycoord) -> (dest_xcoord, dest_ycoord)
-   //
-   // Variables for Bresenham's line algorithm:
-   // x0, x1, y0, y1... (already transformed in hops and bins space}
    if(!mpSpectralData->coordHistory.empty()){
       int src_xcoord = mpSpectralData->coordHistory.back().first;
       int src_ycoord = mpSpectralData->coordHistory.back().second;
-      int srcFreq = PositionToFrequency(wt, 0, src_ycoord, mRect.y, mRect.height);
 
-      int x0 = posToLongLong(src_xcoord) / mpSpectralData->GetHopSize();
-      int y0 = srcFreq / (mpSpectralData->GetSR() / mpSpectralData->GetWindowSize());
+      int h0 = posXToHopNum(src_xcoord);
+      int b0 = posYToFreqBin(src_ycoord);
       int wd = mBrushRadius * 2;
 
-      int dx =  abs(x1-x0), sx = x0<x1 ? 1 : -1;
-      int dy = -abs(y1-y0), sy = y0<y1 ? 1 : -1;
-      int err = dx+dy, err2;
+      int db =  abs(h1-h0), sb = h0<h1 ? 1 : -1;
+      int dh = -abs(b1-b0), sh = b0<b1 ? 1 : -1;
```

Comment:
Did you mix up the intended definitions of db and dh, sb and sh?

Reply:
Maybe the code is correct, but the substitution of h for x and b for y in variable names isn't consistent, and is sometimes reversed, so it is hard to review it.

Reply:
Sorry you are right about this, have been corrected to stay consistent.

Reply:
- [ ] I